tests/experiment/test_results.py

```
@@ -153,6 +153,91 @@ class DummyProcedure(Procedure):
+
+    @staticmethod
+    def _write_rows(result, rows):
+        """Append data rows directly to the results file."""
+        with open(result.data_filename, 'a', encoding=Results.ENCODING) as f:
+            for row in rows:
+                f.write(result.format(row) + Results.LINE_BREAK)
+
+    def test_results_numeric_dtype_preserved_after_reload(self, tmpdir):
+        """Numeric dtypes must be preserved after writing and reloading a Results file."""
+        class DummyProcedure(Procedure):
+            DATA_COLUMNS = ['x', 'y']
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)